src/main/java/org/littil/api/guestTeacher/service/GuestTeacherService.java

```diff
@@ -35,18 +37,25 @@ public class GuestTeacherService {
     private final UserService userService;
     private final UserMapper userMapper;
     private final AuthenticationService authenticationService;
+    private final TokenHelper tokenHelper;
 
 
     public List<GuestTeacher> getTeacherByName(@NonNull final String name) {
-        return repository.findBySurnameLike(name).stream().map(mapper::toDomain).toList();
+        UUID userId = tokenHelper.getCurrentUserId();
```

Comment:
This method now only returns a teacher object which correnspons with the current user. From my understanding, we had to filter the properties of the teacher, not the teacher itself. For example:

- When fetching teacher details which are attached to my user I will get all the properties available
- When fetching a teacher from another user on the platform I only get properties that are public available. We want to mask address properties, for example, they are only needed for coordinates.

Reply:
I have made some changes to apply this feedback.